packages/graphback-keycloak-authz/src/createCrudServiceAuthConfigs.ts

```diff
@@ -0,0 +1,37 @@
+import { CrudServiceAuthConfig } from './definitions';
+import { getEmptyServiceConfig, getOperationsFromRule } from './utils';
+import { ModelAuthConfigs } from '.';
+
+/**
+ * Convert global auth config supplied by the user into a set of configs that can be applied
+ * to each model's CRUD Service
+ * 
+ * Done in this way so that the user facing config can support 
+ * additional features such as owner based auth in the future without severe breaking changes
+ * 
+ * @param models The model configs that contain auth rules for each model
+ */
+export function createCrudServiceAuthConfigs(models: ModelAuthConfigs): CrudServiceAuthConfig {
```

Comment:
Really clever mapping. However should this actually come from user? Not strong opinion but would developers define this rules per model/service based approach - this kinda boils back to what runtime API will look like, but want to understand why we need that transformation

Reply:
I would need to know what was your intention on this one.
Is this just helper and if we want to generate other queries/mutations there are no problems to extend it?

Reply:
It's really just a transform function that allows a developer to define auth rules in a similar way to how other projects do the same thing. That particular format could be extended to support owner based auth and so on. However, maybe we don't really need at as a user could define the configs on a service level. Something like this:

```js
const authConfigs = {
  Task: {
    create: { roles: ['developer']},
    read: { roles: ['developer']},
    update: { roles: ['admin']},
    delete: { roles: ['admin']},
  },
  Note: {
    ...
  }
}
```

Yeah this will probably be easier for a user to use.

Reply:
Transformation might have bugs or people might expect to do some magic. 
Having more implicit way to define stuff it ok:

```
const authConfigs = {
  Task: {
    create: { roles: ['developer']},
    read: { roles: ['developer']},
    update: { roles: ['admin']},
    // Roles are optional
    //delete: { roles: ['admin']},
  },
  Note: {
    ...
  }
}
```

This format is also good to work with later for relations:

```
const authConfigs = {
  Task: {
    create: { roles: ['developer']},
    comment: { roles: ['admin']},
  }
}
```

Reply:
Made edit.